Src/Coravel/Cache/InMemoryCache.cs

```diff
@@ -72,12 +69,22 @@ public void Forget(string key)
             this._keys.Remove(key);
         }
 
-        public void Flush() {
-            foreach(string key in this._keys){
+        public void Flush()
+        {
+            foreach (string key in this._keys)
+            {
                 this._cache.Remove(key);
             }
 
             this._keys.Clear();
         }
+
+        private void EvictionCallback(object key, object value, EvictionReason reason, object state)
+        {
+            if (reason == EvictionReason.Expired || reason == EvictionReason.Capacity || reason == EvictionReason.Removed)
+            {
+                _keys.Remove(key.ToString());
```

Comment:
Just for consistency could you add `this._keys....` ?

Reply:
Made the requested changes, please let me know if there is anything else to be changed :)